File: acousticslib/time_series_2_cross_correlation.py
import numpy as np
import domain_conversions
import linear_spectrum_2_cross_spectral_density

def timeseries2crosscorrelation(x1,x2,dt):
    X1 = domain_conversions.linearspectrum2timeseries(x1,dt)[0]
    X2 = domain_conversions.linearspectrum2timeseries(x2,dt)[0]
    Sxy = cross_spectral_density(X1,X2,dt)[0]
    Rxy, Rxy_t_range = crossspectraldensity2timeseries(Sxy,dt)

    return Rxy, Rxy_t_range

File: acousticslib_matlab/timeseries2crosscorrelation.m
function [Rxy, Rxy_t_range] = timeseries2crosscorrelation(x1, x2, dt)
[X1, ~] = timeseries2linearspectrum(x1, dt);
[X2, ~] = timeseries2linearspectrum(x2, dt);
[Sxy, ~, ~] = linearspectrum2crossspectraldensity(X1, X2, dt);
[Rxy, Rxy_t_range] = crossspectraldensity2timeseries(Sxy, dt);
end